src/Provider/Photon/Model/PhotonAddress.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Geocoder package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @license    MIT License
+ */
+
+namespace Geocoder\Provider\Photon\Model;
+
+use Geocoder\Model\Address;
+
+/**
+ * @author Jonathan Beliën <jbe@geo6.be>
+ */
+final class PhotonAddress extends Address
+{
+    /**
+     * @var int|null
+     */
+    private $osmId;
+
+    /**
+     * @var string|null
+     */
+    private $osmType;
+
+    /**
+     * @var \stdclass|null
+     */
+    private $osmTag;
+
+    /**
+     * @return null|int
+     */
+    public function getOSMId()
+    {
+        return $this->osmId;
+    }
+
+    /**
+     * @param null|int $osmId
+     *
+     * @return PhotonAddress
+     */
+    public function withOSMId(int $osmId = null): self
```

Comment:
What are you think about to use this structure `withOSMId(?int $osmId)`.

With current structure I can call with no pass argument `withOSMId()`.

Reply:
That would be better indeed but Geocoder is suppose to be compatible with PHP 7.0 (see [here](https://github.com/geocoder-php/Geocoder/blob/master/composer.json#L15)) and [Nullable types](http://php.net/manual/en/migration71.new-features.php#migration71.new-features.nullable-types) is a PHP 7.1 feature.